documentation/directory-structure.md

```diff
@@ -0,0 +1,197 @@
+
+`rpnamespace/stable/api_version`  
+`rpnamespace/preview/api_version`
+
+If the contents of `data-plane` folder are not organized by `rpnamespace`, we treat
+the `data-plane` folder itself as the "default namespace" for the purposes of describing
+the required layout.
+
+## API version (`api_version`) folders layout
+
+`api_version` within given `rpnamespace/stable` or `rpnamespace/preview` is a versioned set of OpenAPI specs. `stable` and `preview` correspond to the lifecycle stages of the API versions within. Learn more about API versions, their directory layout and their lifecycle stages at [aka.ms/azsdk/pr-api-versions].
+
+Each `api_version` folder is a direct child of the `preview` or `stable` folder. This folder contains the REST OpenAPI Specs `.json` files, and the `examples` folder.
+
+### `examples` folder
+
+Each `api_version/examples` folder contains [`x-ms-examples`] files.
+
+## Naming guidelines for `specification` folder contents
+
+- Folder names should be singular (e.g. `keyvault` not `keyvaults` ) -- this removes ambiguity for some non-english speakers.
+- Generic folder names should be lower-case.
+- Namespace (`rpnamespace`) folders can be PascalCased (e.g. `KeyVault`).
+- Files are whatever case you think is good for your soul.
+
+## Example layout of a `specification` folder
+
+```bash
+.
+\---specification
+|    +---automation
+|    |   \---resource-manager
+|    |       +---Microsoft.Automation
+|    |       |   \---stable
+|    |       |       \---2015-10-31
+|    |       |           \---examples
+|    |       \---readme.md
+|    +---batch
+|    |   +---data-plane
+|    |   |   +---stable
+|    |   |   |   +---2016-07-01
+|    |   |   |   |   \---examples
+|    |   |   |   \---2017-01-01
+|    |   |   |       \---examples
+|    |   |   +---preview
+|    |   |   |   \---2017-05-01-preview
+|    |   |   |       \---examples
+|    |   |   \---readme.md
+|    |   \---resource-manager
+|    |       +---Microsoft.Batch
+|    |       |   \---stable
+|    |       |       +---2015-12-01
+|    |       |       |   \---examples
+|    |       |       +---2017-01-01
+|    |       |       |   \---examples
+|    |       |       \---2017-05-01
+|    |       |           \---examples
+|    |       \---readme.md
+|    \---playfab
+|        +---Playfab
+|        |   \---tspconfig.yaml
+|        |   \---main.tsp
+|        \--resource-manager
+|            +---Microsoft.Playfab
+|            |   \---preview
+|            |       \---2017-04-24-preview
+|            |           \---examples
+|            \---readme.md
+```
+
+In the example above:
+
+`rp` folders are `automation`, `batch`, `playfab`.  
+TypeSpec folders are `specification/playfab/Playfab`.  
+`rpnamespace` names present are: `Microsoft.Automation` (in `automation` rp), `Microsoft.Batch` (in `batch` rp) and `Microsoft.Playfab` (in `playfab` rp).
+
+## Folder Structure for service groups
```

Comment:
@JeffreyRichter I significantly shortened this section and added a note of this not being allowed going forward. Can you review?

Reply:
Lets setup a meeting and discuss.